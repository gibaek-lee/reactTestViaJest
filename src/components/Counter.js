import React from 'react';
import PropTypes from 'prop-types';

const Counter = ({value, onIncrease, onDecrease}) => {
  return (
    <div>
      <h1>Counter</h1>
      <h2>{value}</h2>
      <button onClick={onIncrease}>+</button>
      <button onClick={onDecrease}>-</button>
    </div>
  );
};
Counter.propTypes = {
  value: PropTypes.number,
  onIncrease: PropTypes.func,
  onDecrease: PropTypes.func
}
Counter.defaultProps = {
  value: 1,
  onIncrease: () => console.warn('onIncrease not defined'),
  onDecrease: () => console.warn('onDecrease not defined')
}

export default Counter;
