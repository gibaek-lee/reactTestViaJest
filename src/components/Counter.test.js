import React from 'react';
import { shallow } from 'enzyme';
import Counter from './Counter';

describe('Counter', () => {//describe
  let component = null;
  //jest mock functions(https:jestjs.io/docs/en/mock-functions.html)
  const mockIncrease = jest.fn();
  const mockDecrease = jest.fn();

  /* component initial render test */
  it('renders correctly', () => {
    component = shallow(<Counter value={700} onIncrease={mockIncrease} onDecrease={mockDecrease}/>);
  });

  /* snapshot test */
  it('matches snapshot', () => {//initial rendering snapshot matching test
    expect(component).toMatchSnapshot();
  })

  /* redux + react: props.value 전달 가정한 rendering test*/
  it('is 700', () => {
    expect(component.find('h2').text()).toBe('700');
  });

  /* redux + react: props.increase() props.decrease() 전달 가정하고, 호출됐을 때 호출여부 test */
  it('calls functions', () => {
    const buttons = component.find('button');
    buttons.at(0).simulate('click');
    buttons.at(1).simulate('click');
    expect(mockIncrease.mock.calls.length).toBe(1);
    expect(mockDecrease.mock.calls.length).toBe(1);
  });

});
