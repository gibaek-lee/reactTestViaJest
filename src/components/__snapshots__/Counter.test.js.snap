// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches snapshot 1`] = `
<div>
  <h1>
    Counter
  </h1>
  <h2>
    700
  </h2>
  <button
    onClick={[MockFunction]}
  >
    +
  </button>
  <button
    onClick={[MockFunction]}
  >
    -
  </button>
</div>
`;
